### ISSUE TITLE
'%AppData' returned empty string, error message ""

### ISSUE THREAD

**Issue by Supreeeme** [Winetricks/winetricks#1647]
On Arch Linux & Wine 5.21, tried with both winetricks from the main repos and winetricks-git from the AUR, as well as wine staging. This is a fresh prefix.

``` 
~  winetricks                      
Executing mkdir -p /home/supreme
------------------------------------------------------
warning: You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug.
------------------------------------------------------
------------------------------------------------------
WINEPREFIX INFO:
Drive C: total 32
drwxr-xr-x  7 supreme supreme 4096 Nov 18 14:21 .
drwxr-xr-x  4 supreme supreme 4096 Nov 18 14:23 ..
drwxr-xr-x  3 supreme supreme 4096 Nov 18 14:12 ProgramData
drwxr-xr-x  6 supreme supreme 4096 Nov 18 14:12 Program Files
drwxr-xr-x  6 supreme supreme 4096 Nov 18 14:12 Program Files (x86)
drwxr-xr-x  4 supreme supreme 4096 Nov 18 14:12 users
drwxr-xr-x 19 supreme supreme 4096 Nov 18 14:12 windows
-rw-r--r--  1 supreme supreme    8 Nov 18 14:21 .windows-serial

Registry info:
/home/supreme/.wine/system.reg:#arch=win64
/home/supreme/.wine/userdef.reg:#arch=win64
/home/supreme/.wine/user.reg:#arch=win64
------------------------------------------------------
------------------------------------------------------
warning: wine cmd.exe /c echo '%AppData%' returned empty string, error message "" 
------------------------------------------------------
```
But that doesn't seem true at all:
```
~  wine cmd.exe /c echo '%AppData%'
C:\users\supreme\Application Data
```

**Comment by austin987** [comment]
There was a race in 5.21 you might've been seeing (https://bugs.winehq.org/show_bug.cgi?id=50046).

I'd suggest trying wine-6.0-rc1 (and winetricks git).

**Comment by Supreeeme** [comment]
after compiling 6.0 winetricks is fine now, great.

**Comment by noah1510** [comment]
I still get the problem.
My wine version is 6.2-staging and I use winetricks 20210206-next.
Is this a problem that is caused by a 64-bit wine prefix?
Even after deleting `~/.wine` and letting a new one be created the problem exists.

**Comment by austin987** [comment]
This issue is closed, you should open your open issue.

That said, I can't reproduce, so someone that can will likely need to debug it.

**Comment by austin987** [comment]
For Debian (specifically 8.0~repack-4, though maybe also 7.22-repack-2), that's https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1031649 (see also https://github.com/Winetricks/winetricks/issues/2030)

**Comment by joel-sitte** [comment]
I am also seeing this on Ubuntu.

wine --version returns:
wine-8.0 (Ubuntu 8.0~repack-4)

winetricks --version returns:
20230212-next - sha256sum: fe0550e0d843214f87dcb0f4aa591be0046fa93db7b8330217dd926258e628fc

Does this bug need to be re-opened? or do we need to open a new instance of this bug?

**Comment by joel-sitte** [comment]
Also, I am willing to try some debugging when I have time but would need some pointers on how best to do that.

**Comment by austin987** [comment]
I just tested installing a Ubuntu 23.04 VM, then installed wine from the repo (Ubuntu 8.0~repack-4), and downloaded winetricks. I can reproduce the error.

If I compile wine myself, however, it works fine.

This should be reported to Debian/Ubuntu. While it may be possible to work around it in Winetricks, I don't have time to debug distro specific issues.

Note for upstream: I noticed that /usr/bin/wine is actually a shell script, which may be part of the issue. That will not work in future Wine versions, see https://bugs.winehq.org/show_bug.cgi?id=54397#c4
